src/main/java/com/mael/ttt/Game.java

```diff
@@ -53,8 +51,8 @@ private void setPlayers() {
     private boolean nextTurn() {
         gameUI.printBoard(board);
 
-        currentPlayer = (currentMark.equals("X")) ? player1 : player2;
-        int index     = currentPlayer.getCellIndex(board);
+        Player currentPlayer = (currentMark.equals("X")) ? player1 : player2;
+        int index            = currentPlayer.getCellIndex(board);
```

Comment:
This code looks like there is a very hard coupling between a `String` representation of something on the board and the relationship it has towards the logic of deciding who's turn it is.